Userland/Services/AudioServer/Mixer.cpp

```diff
@@ -65,13 +66,12 @@ void Mixer::mix()
     decltype(m_pending_mixing) active_mix_queues;
 
     for (;;) {
-        if (active_mix_queues.is_empty() || m_added_queue) {
-            pthread_mutex_lock(&m_pending_mutex);
-            pthread_cond_wait(&m_pending_cond, &m_pending_mutex);
+        m_pending_mutex.lock();
+        // While we have nothing to mix, wait on the condition.
+        m_mixing_necessary.wait_while([this, &active_mix_queues]() { return m_pending_mixing.is_empty() && active_mix_queues.is_empty(); });
+        if (!m_pending_mixing.is_empty())
             active_mix_queues.extend(move(m_pending_mixing));
-            pthread_mutex_unlock(&m_pending_mutex);
```

Comment:
Can we really rely on the `move()` clearing the vector? I'd rather we clear it explicitly after the move.